server/src-lib/Control/Concurrent/Extended.hs

```diff
@@ -8,9 +10,15 @@ import           Prelude
 import qualified Control.Concurrent    as Base
 
 import           Control.Concurrent    hiding (threadDelay)
-import           Data.Time.Clock       (DiffTime)
-import           Data.Time.Clock.Units (Microseconds (..))
+import           Data.Time.Clock.Units (Microseconds (..), NominalDiffTime)
 
--- | Like 'Base.threadDelay', but takes a 'DiffTime' instead of an 'Int'.
-threadDelay :: DiffTime -> IO ()
-threadDelay = Base.threadDelay . round . Microseconds
+-- | Like 'Base.threadDelay', but takes a 'NominalDiffTime' instead of an 'Int' microseconds.
+--
+-- NOTE: you cannot simply replace e.g. @threadDelay 1000@ with @sleep 1000@ since those literals
+-- have different meanings!
+sleep :: NominalDiffTime -> IO ()
+sleep = Base.threadDelay . round . Microseconds
+
+{-# DEPRECATED threadDelay "Please use `sleep` instead (and read the docs!)" #-}
+threadDelay :: Int -> IO ()
+threadDelay = Base.threadDelay
```

Comment:
I understand the intent here (of guiding people who might have used `threadDelay` towards the new thing), but it seems a little curious to have a `DEPRECATED` binding in our own application code. Maybe it’s fine, though, I don’t have enormously strong feelings about it.

Reply:
@jberryman Might as well get rid of `threadDelay` in the codebase?

Reply:
In the last call I tried to outline the pattern. When (if) eventually this `Extended` module can shadow the standard module I think we should really strive to keep the changes and tweaks as obvious and discoverable as possible. I anticipate more `threadDelay` being written before the mixin shadowing thing happens, then all of these will become visible with a warning. I think there's no downside to keeping this in since we're fairly disciplined about keeping the haskell code warning free.

If I can do the mixin shadowing pattern successfully (now that we've moved to cabal-install) I'll write up a blog about the pattern and articulate thoughts a little more clearly and get feedback.